### ISSUE TITLE
[2.19] `DateTime`'s new `copyWith` method

### ISSUE THREAD

**Issue by MaryaBelanger** [dart-lang/site-www#4318]
### This is an _evaluation_ issue
Evaluating the _possible_ need for doc updates due to https://github.com/dart-lang/sdk/issues/24644

### Evaluation
Maybe add something to this section: https://dart.dev/guides/libraries/library-tour#dates-and-times
Though that section doesn't necessarily cover every `DateTime` method, so maybe just the API docs are sufficient

**Comment by mit-mit** [comment]
@lrhn we'd like to add a simple example of this to https://dart.dev/guides/libraries/library-tour#dates-and-times. wdyt about adding some like the last two lines here?

```dart
// Get the current date and time.
var now = DateTime.now();

// Create a new DateTime with the local time zone.
var y2k = DateTime(2000); // January 1, 2000
...
// Parse an ISO 8601 date.
y2k = DateTime.parse('2000-01-01T00:00:00Z');

// Create a new DateTime from an existing one, adjusting just some properties:
var nowLastYear = now.copyWith(year: now.year-1);
```

**Comment by lrhn** [comment]
LGTM.
Nits:
```dart
// Parse an ISO 8601 date in the UTC time zone.
y2k = DateTime.parse('2000-01-01T00:00:00Z');

// Create a new `DateTime` from an existing one, adjusting just some properties:
var sameTimeLastYear = now.copyWith(year: now.year - 1);
```

**Comment by MaryaBelanger** [comment]
Fixing PRs merged into [v2.19](https://github.com/dart-lang/site-www/tree/v2.19)
